src/Microsoft.DotNet.Build.Tasks/PackageFiles/Build.Common.targets

```diff
@@ -106,7 +106,7 @@
        the targets necessary to build tests with project references converted to package dependencies.
   -->
   <PropertyGroup>
-    <_BuildAgainstPackages Condition="'$(KeepAllProjectReferences)' != 'true' and '$(IsTestProject)' == 'true' and '$(MSBuildProjectExtension)' != '.builds'">$(BuildTestsAgainstPackages)</_BuildAgainstPackages>
+    <_BuildAgainstPackages Condition="'$(KeepAllProjectReferences)' != 'true' and '$(IsTestProject)' == 'true' and '$(MSBuildProjectExtension)' != '.builds' and '$(SkipBuildAgainstPackages)'=='true'">true</_BuildAgainstPackages>
```

Comment:
I wouldn't expect `SkipBuildAgainstPackages=true` to set `_BuildAgainstPackages` to "true"